bin/varnishd/cache/cache_fetch.c

```diff
@@ -483,19 +482,39 @@ vbf_stp_startfetch(struct worker *wrk, struct busyobj *bo)
 	AZ(bo->do_esi);
 	AZ(bo->was_304);
 
-	if (http_IsStatus(bo->beresp, 304)) {
-		if (!vbf_expected_304(bo))
-			return (F_STP_ERROR);
-		else if (bo->stale_oc != NULL &&
-		    ObjCheckFlag(bo->wrk, bo->stale_oc, OF_IMSCAND))
-			vbf_304_logic(bo);
+	if (bo->stale_oc != NULL) {
+		if (http_IsStatus(bo->beresp, 304) &&
+		    vbf_expected_304(bo))
+			bo->was_304 = 1;
+		VCL_backend_refresh_method(bo->vcl, wrk, NULL, bo, NULL);
+		switch (wrk->vpi->handling) {
+				case VCL_RET_MERGE:
+					vbf_304_logic(bo);
+					break;
+				case VCL_RET_BERESP:
+					break;
+				case VCL_RET_OBJ_STALE:
+					HTTP_Decode(bo->beresp, ObjGetAttr(bo->wrk, bo->stale_oc, OA_HEADERS, NULL));
+					break;
+				case VCL_RET_ERROR:
+					/* FALLTHROUGH */
+				case VCL_RET_ABANDON:
+					/* FALLTHROUGH */
+				case VCL_RET_FAIL:
+					/* FALLTHROUGH */
+				case VCL_RET_RETRY:
+					skip_vbr = 1;
+					break;
```

Comment:
You can simply do this:

```c
case VCL_RET_ERROR:
case VCL_RET_ABANDON:
case VCL_RET_FAIL:
case VCL_RET_RETRY:
	skip_vbr = 1;
	break;
```

It shouldn't trigger a warning.

Reply:
I remember doing that on another PR and getting CCI failing compilation because of `-Wimplicit-fallthrough`

Reply:
I think that only happens if you got statements in-between without a break.

Reply:
Oh right, I can test it then